packages/core/lib/commands/debug/run.js

```diff
@@ -60,9 +60,20 @@ module.exports = async function (options) {
   if (config.compileAll && config.compileNone) {
     throw new Error("Incompatible options passed regarding what to compile");
   }
-  const interpreter = await new CLIDebugger(config, {
+
+  // Create a new CLIDebugger instance
+  const cliDebugger = new CLIDebugger(config, {
     txHash,
     compilations
-  }).run();
+  });
+
+  // Checks if the user wants to open the debugger in vscode
+  if (config.vscode) {
+    await cliDebugger.openVSCodeDebug();
```

Comment:
Shouldn't this method not be part of the thing called the "CLI" debugger? Seems better to keep the CLIDebugger just for the CLI and push this VS Code logic elsewhere.

Reply:
Hi @gnidan, Thank you for that

what place do you suggest?

Reply:
@kevinbluer for visibility

Reply:
Thanks for the heads up @xhulz! Also sounds like we've got a way forward for this from our earlier conversation 👍

Reply:
Hm, maybe just have a `VSCodeDebugger` class?

Reply:
Hey @gnidan @kevinbluer 

I just moved all the vscode debug function to a new class called `VSCodeDebugger`.

Thank you all again